src/main/java/tictactoe/ComputerPlayer.java

```diff
@@ -20,78 +22,114 @@ public final int nextMove() {
         int bestMoveEvaluated = negativeInfinity;
         int alpha = negativeInfinity;
         int beta = positiveInfinity;
+        int depth = 40;
 
         for (Integer position : board.freePosition()) {
-            int valueMove = evaluateMove(board, this.mark, alpha, beta, position);
+            int valueMove = evaluateMove(board, this.mark, alpha, beta, position, depth, System.nanoTime());
 
             if (valueMove > bestMoveEvaluated) {
                 bestMoveEvaluated = valueMove;
                 bestPosition = position;
             }
+
+            if (perfectMoveFound(bestMoveEvaluated)) {
+                break;
+            }
+
         }
 
-        return bestPosition;
+        if (!couldEvaluateMove(bestMoveEvaluated)) {
+            return randomMove();
+        } else {
+            return bestPosition;
+        }
+    }
+
+    private boolean perfectMoveFound(int bestMoveEvaluated) {
+        return bestMoveEvaluated == 1;
+    }
+
+    private boolean couldEvaluateMove(int bestMoveEvaluated) {
+        return bestMoveEvaluated != -2;
```

Comment:
This use of the `valueMove` variable feels odd.
Is it actually `theValueOfPossiblyTheBestMove`? Or Is it `theValueOfThePossiblyBestMoveUnlessItsTwoThenItsTheFactThatICouldNotAnswerFastEnough`?

If it is the second case, consider creating a small class, `ValuedMove` that captures the notion of "A move that has a value, but might also not have been evaluated" and has has meaningful methods and fields and tests... Micro-Types are OK. They are pretty cool. They give meaning and intent to simple things as "int or null" 😄